astropy/config/paths.py

```diff
@@ -19,6 +20,46 @@
 ]
 
 
+class _DirectoryType(Enum):
+    CONFIG = auto()
+    CACHE = auto()
+
+
+def _get_dir_path(rootname: str, dirtype: _DirectoryType) -> Path:
+    match dirtype:
+        case _DirectoryType.CONFIG:
+            cls = set_temp_config
+            fallback = "config"
+        case _DirectoryType.CACHE:
+            cls = set_temp_cache
+            fallback = "cache"
+        case _:
+            raise ValueError(f"Unexpected {dirtype=}")
+
+    # If using set_temp_x, that overrides all
+    if cls._temp_path is not None:
```

Comment:
Absolute nit, but a good opportunity for the walrus: `if (xch := cls._tmp_path) is not None:` (and nice for symmetry with the below).